core/runtime/src/main/java/io/quarkus/runtime/logging/LogMetricsHandlerRecorder.java

```diff
@@ -0,0 +1,55 @@
+package io.quarkus.runtime.logging;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Consumer;
+import java.util.logging.Handler;
+
+import org.jboss.logmanager.Level;
+
+import io.quarkus.runtime.RuntimeValue;
+import io.quarkus.runtime.annotations.Recorder;
+import io.quarkus.runtime.metrics.MetricsFactory;
+
+@Recorder
+public class LogMetricsHandlerRecorder {
+
+    static final String METRIC_NAME = "log.total";
+
+    static final String METRIC_DESCRIPTION = "Number of log events";
+
+    static final List<Level> STANDARD_LEVELS = Arrays.asList(Level.FATAL, Level.ERROR, Level.WARN, Level.INFO, Level.DEBUG,
+            Level.TRACE);
+
+    Map<Level, AtomicLong> logCounters = new HashMap<>();
```

Comment:
This should be `final`.

Reply:
In addition, I would map as many non-standard levels as possible to these counters, for example:

```java
    logCounters.put(java.util.logging.Level.WARNING, logCounters.get(Level.WARN));
```

This will increase the accuracy of the counts.  Note that level equivalency should be determined by the level's integer value.

Reply:
Hmm I can't seem to put it final, it's always generating a setter which fails to compile (/LogMetricsHandlerRecorder$$accessor.java:[10,48] cannot assign a value to final variable logCounters

Reply:
Ok had to also be static.
I've used level integer as key to also catch any equivalent levels.